extensions/wikia/AdEngine/js/template/bfaaDesktop.js

```diff
@@ -72,20 +71,37 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 			recoveryTweaker.tweakSlot(params.slotName, iframe);
 		}
 
-		if (params.videoUrl && params.videoTriggerElement) {
-			slotTweaker.onReady(params.slotName, function() {
-				var divs = doc.querySelectorAll('#' + params.slotName + ' > div'),
-					imageContainer = divs[divs.length - 1],
-					video = uapVideoAd.init(doc.getElementById(params.slotName), imageContainer, params.videoUrl);
-
-				params.videoTriggerElement.addEventListener('click', function () {
-					uapVideoAd.playAndToggle(video, imageContainer);
-				});
+		if (params.videoTriggerElement && params.videoAspectRatio) {
+			videoAdFactory.init().then(function () {
```

Comment:
How about extracting the anonymous function outside, so the `runOnReady()` is more readable and we remove so many indentations?